xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -1371,3 +1372,35 @@ time_t CPVRClients::GetBufferTimeEnd() const
 
   return time;
 }
+
+bool CPVRClients::NextEventWithinBackendIdleTime(const CPVRTimers& timers)
+{
+    // timers going off soon?
+    const CDateTime now(CDateTime::GetUTCDateTime());
+    const CDateTimeSpan idle(
+      0, 0, CSettings::Get().GetInt("pvrpowermanagement.backendidletime"), 0);
+    const CDateTime next(timers.GetNextEventTime());
+    const CDateTimeSpan delta(next - now);
+
+    return (delta <= idle);
+}
+
+bool CPVRClients::AllLocalBackendsIdle() const
+{
+  PVR_CLIENTMAP clients;
+  GetConnectedClients(clients);
+  for (PVR_CLIENTMAP_CITR itr = clients.begin(); itr != clients.end(); itr++)
```

Comment:
@xhaggi : LOL. Actually I copied the code for the loop.  It is done like this all over the place in CPVRClients. But yes <sigh>, will change it, hoping that @opdenkamp has no problem with this.